src/main/java/uk/gov/companieshouse/itemhandler/kafka/OrdersKafkaConsumer.java

```diff
@@ -0,0 +1,137 @@
+    private static final Logger LOGGER = LoggerFactory.getLogger(APPLICATION_NAMESPACE);
+    private static final String ORDER_RECEIVED_TOPIC = "order-received";
+    private static final String ORDER_RECEIVED_TOPIC_RETRY = "order-received-retry";
+    private static final String ORDER_RECEIVED_TOPIC_ERROR = "order-received-error";
+    private static final int MAX_RETRY_ATTEMPTS = 3;
+    private static final String GROUP_NAME = "order-received-consumers";
+    private CHKafkaResilientConsumerGroup chKafkaConsumerGroupMain;
+    private CHKafkaResilientConsumerGroup chKafkaConsumerGroupRetry;
+    private CHKafkaResilientConsumerGroup chKafkaConsumerGroupError;
+    @Value("${kafka.broker.addresses}")
+    private String brokerAddresses;
+    private CHKafkaProducer chKafkaProducer;
+    private final SerializerFactory serializerFactory;
+
+    public OrdersKafkaConsumer(SerializerFactory serializerFactory) {
+        this.serializerFactory = serializerFactory;
+    }
+
+    @Override
+    public void afterPropertiesSet() throws Exception {
+        LOGGER.debug("Initializing kafka consumer service " + this.toString());
+
+        ProducerConfig config = getProducerConfig();
+        chKafkaProducer = new CHKafkaProducer(config);
+
+        ConsumerConfig consumerConfig = getConsumerConfig();
+
+        chKafkaConsumerGroupMain = new CHKafkaResilientConsumerGroup(consumerConfig, CHConsumerType.MAIN_CONSUMER,
+                new KafkaConsumerFactory(), chKafkaProducer);
+        chKafkaConsumerGroupRetry = new CHKafkaResilientConsumerGroup(consumerConfig, CHConsumerType.RETRY_CONSUMER,
+                new KafkaConsumerFactory(), chKafkaProducer);
+    }
+
+    @KafkaListener(topics = ORDER_RECEIVED_TOPIC, groupId = GROUP_NAME)
+    public void processOrderReceived(String orderReceivedUri)
+            throws SerializationException, ExecutionException, InterruptedException {
+        try {
+            LOGGER.info("Message: " + orderReceivedUri + " received on topic: " + ORDER_RECEIVED_TOPIC);
+        } catch (Exception x){
+            Message message = createRetryMessage(orderReceivedUri);
+            for (int attempt = 0; attempt < MAX_RETRY_ATTEMPTS; attempt++) {
+                chKafkaConsumerGroupMain.retry(attempt, message);
+            }
+        }
+    }
+
+    @KafkaListener(topics = ORDER_RECEIVED_TOPIC_RETRY, groupId = GROUP_NAME)
+    public void processOrderReceivedRetry(String orderReceivedUri)
+            throws SerializationException, ExecutionException, InterruptedException {
+        try {
+            LOGGER.info("Message: " + orderReceivedUri + " received on topic: " + ORDER_RECEIVED_TOPIC_RETRY);
+        } catch (Exception x){
+            Message message = createRetryMessage(orderReceivedUri);
+            for (int attempt = 0; attempt < MAX_RETRY_ATTEMPTS; attempt++) {
+                chKafkaConsumerGroupRetry.retry(attempt, message);
+            }
+        }
+    }
+
```

Comment:
- Do we not need a listener/consumer for the error topic (`ORDER_RECEIVED_TOPIC_ERROR `) too?
- As I noted in the technical requirement referred to in an earlier comment, `chips-filing-consumer` has a separate application instance for the error topic consumer. See: https://github.com/companieshouse/chips-filing-consumer/blob/develop/src/main/java/uk/gov/ch/Main.java#L110-L138. Is this generally required, or something specific to that application?
- I note that the `ThreadedMessageConsumer` seems to be part of the way the library is expected to be used. Have you ascertained whether your approach is equivalent?

Reply:
* implemented a listener for `ORDER_RECEIVED_TOPIC_ERROR`
* The way `chips-filing-consumer` handles reading messages from error topics is by using a specific consumer group run on a Thread based consumer. This implementation follows a similar approach - using a specific consumer group for each of the topics `-main`, `-retry` and `-error`. Implementing these consumers using `@KafkaListener` gives the advantage of separating the handling cleanly without having to manage consumer concurrency.
* `@KafkaListener` offers all of the benefits of `ThreadedMessageConsumer` with more granular configurability that a resilient listener needs. This implementation uses a hybrid approach - allowing consumer configuration to be managed by CHKafka library as well as Spring. 
* https://docs.spring.io/autorepo/docs/spring-kafka-dist/1.1.7.BUILD-SNAPSHOT/reference/htmlsingle/#__kafkalistener_annotation

Reply:
* We have an error topic listener now - very good.
* It still seems to me however that we should consider why `chips-filing-consumer` has a separate process for picking up messages from the `error` topic. I think the answer may be in the `Error topic` slide of the `kafka-resilience-with-detail` PowerPoint presentation that Joanne gave us: 

> If the max retry attempts have failed then the email request is moved to the error topic. This topic is not consumed by the normal running application. Once the issue has been fixed then the emails can be re-processed using an instance of the application run with the error consumer flag set.

Reply:
Implemented a property to control which listeners start when the application boots. This defaults to `false` enabling both `-main` and `-retry` listeners in normal mode.

Reply:
* I don't think this gives the equivalent degree of control and flexibility as the environment variable based approach seen in `chips-filing-consumer` - and a number of other consumers in CHS. The approach they seem to follow also has the advantage of following a consistent pattern, which could reduce the learning curve and associated support risks? See the READMEs of CHS consumers, a number of which refer to the `IS_ERROR_QUEUE_CONSUMER` environment variable.  
* Do we have automated tests verifying the impact of this property/variable?

Reply:
* The property `uk.gov.companieshouse.item-handler.error-consumer` now picks up its value from `IS_ERROR_QUEUE_CONSUMER` env variable.
* There are two sets of integration tests that demonstrate the effect of enabling/disabling this property.